### PR TITLE
GI-137: Quiz Layout Redesign

### PR DESCRIPTION
**Pull request by psophis**
## Description

- Add mascot (level 2)
- Fix button layout
- Add home button
- Add dialogue when home button is clicked
- Add global components for primary button, secondary button and quiz card

Mobile View:

https://github.com/user-attachments/assets/7d8676bc-c7b7-42e6-aa7c-8861614128e4

Desktop View:

https://github.com/user-attachments/assets/d74e0889-3246-40ad-b889-b2ae26525598

## Jira Ticket

Link to the Jira ticket: [GI-137](https://techstudyfinder.atlassian.net/browse/GI-137)

## Author Checklist

- [x] Code follows project coding standards and conventions
- [x] Tests have been written or updated
- [x] All tests pass locally
- [x] No sensitive data or secrets are included
- [x] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

---

_Please ensure all criteria are met before merging._

### CODE REVIEW

File: client/src/layouts/QuizLayout.tsx [hunk 1]
```diff
@@ -38,57 +40,92 @@ const QuizLayout = ({
...
+          <Box
+            sx={{
+              display: "grid",
+              gap: 2,
+              gridTemplateColumns: "1fr auto auto",
+              alignItems: "center",
+              mb: 4,
+            }}
+          >
+            <BackButton />
```
[the remark this answers is not in the document]
Comment (johanna2402) [reply client/src/layouts/QuizLayout.tsx]: Hier einmal showBackbutton reinmachen damit er nicht im ersten Level ist. Und eine Box in der Mitte damit der Homebutton rechts bleibt.

File: client/src/layouts/QuizLayout.tsx [hunk 2]
```diff
@@ -38,57 +40,95 @@ const QuizLayout = ({
...
+              alignItems: "center",
+              mb: 4,
+            }}
+          >
+              {_showBackButton && (
+            <BackButton label="Zurück" onClick={_oneBack} sx={{}} />
+              )}
+                <Box />
+            <HomeButton onClick={() => setDialogOpen(true)} />
+          </Box>
```
[the remark this answers is not in the document]
Comment (psophis) [reply client/src/layouts/QuizLayout.tsx]: Aber der soll doch links sein?
